File: AndroidTutorial/src/ecologylab/chatTutorial/ChatRequest.java
package ecologylab.chatTutorial;

import java.util.HashMap;

import ecologylab.collections.Scope;
import ecologylab.oodss.distributed.common.SessionObjects;
import ecologylab.oodss.distributed.server.clientsessionmanager.SessionHandle;
import ecologylab.oodss.messages.OkResponse;
import ecologylab.oodss.messages.RequestMessage;
import ecologylab.oodss.messages.ResponseMessage;
import ecologylab.serialization.annotations.simpl_scalar;

public class ChatRequest extends RequestMessage
{
	/**
	 * The message being posted.
	 */
	@simpl_scalar
	String									message;

	/**
	 * Constructor used on server. Fields populated automatically by
	 * ecologylab.serialization
	 */
	public ChatRequest()
	{
	}

	/**
	 * Constructor used on client.
	 * 
	 * @param newEcho
	 *           a String that will be passed to the server to broadcast
	 * */
	public ChatRequest(String message)
	{
		this.message = message;
	}

	/**
	 * Called automatically by LSDCS on server
	 */
	@Override
	public ResponseMessage performService(Scope cSScope)
	{
		//implemented in ChatRequest on C# server side;
		/*
		 * Send back a response confirming that we got the request 
		 */
		return OkResponse.reusableInstance;
	}

	public boolean isDisposable()
	{
		return true;
	}
}

File: DotNetTutorial/Chat/ChatRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Simpl.Fundamental.Generic;
using Simpl.OODSS.Distributed.Common;
using Simpl.OODSS.Messages;
using Simpl.OODSS.Distributed.Server.ClientSessionManager;
using Simpl.Serialization.Attributes;
using ecologylab.collections;

namespace WebSocketOODSSTutorial.Chat
{
    class ChatRequest:RequestMessage
    {
        [SimplScalar] private string message;

        public ChatRequest()
        {
        }

        public ChatRequest(string message)
        {
            this.message = message;
        }

        public override ResponseMessage PerformService(Scope<object> clientSessionScope)
        {
            var sessionManagerMap =
                (DictionaryList<object, WebSocketClientSessionManager>)
                clientSessionScope.Get(SessionObjects.SessionsMap);

            string sessionId = (string) clientSessionScope.Get(SessionObjects.SessionId);

            ChatUpdate messageUpdate = new ChatUpdate(message, sessionId);

            foreach (WebSocketClientSessionManager otherClient in sessionManagerMap.Values)
            {
                Console.WriteLine(otherClient.SessionId + "Checking to make sure not " + sessionId);
                if (!otherClient.SessionId.Equals(sessionId))
                {
                    otherClient.SendUpdateToClient(messageUpdate, otherClient.SessionId);
                }
            }

            return OkResponse.ReusableInstance;
        }
    }
}
